source/org/zfin/expression/repository/HibernateExpressionRepository.java

```diff
@@ -59,1296 +59,1283 @@
-	}
-
-	/**
-	 * Delete a structure from the pile.
-	 *
-	 * @param structure expression structure
-	 */
-	public void deleteExpressionStructure(ExpressionStructure structure) {
-		Session session = HibernateUtil.currentSession();
-		session.delete(structure);
-	}
-
-	public void deleteExpressionStructuresForPub(Publication publication) {
-		List<ExpressionStructure> structures = HibernateUtil.currentSession()
-			.createCriteria(ExpressionStructure.class)
-			.add(Restrictions.eq("publication", publication))
-			.list();
-		for (ExpressionStructure structure : structures) {
-			getInfrastructureRepository().deleteActiveDataByZdbID(structure.getZdbID());
-		}
-	}
-
-	/**
-	 * Delete an expression result record for a given figure.
-	 * If the result has more than one figure it only removes the figure-result association.
-	 * It removes the result object in case there is only one matching figure or if no figure is associated
-	 * and adds an 'unspecified' term to the efs.
-	 *
-	 * @param result expression result.
-	 * @param figure Figure
-	 */
-	public void deleteExpressionResultPerFigure(ExpressionResult2 result, Figure figure) {
+                        session.flush();
+                    }
+                    session.save(result);
+                    result.getExpressionExperiment().addExpressionResult(result);
+                } else {
+                    // has more than one figure associated
+                    unspecifiedResult.removeFigure(singleFigure);
+                    session.save(result);
+                    result.getExpressionExperiment().addExpressionResult(result);
+                }
+            } else {
+                session.save(result);
+                result.getExpressionExperiment().addExpressionResult(result);
+            }
+            runAntibodyAnatomyFastSearchUpdate(result);
+        }
+    }
+*/
+
+    private ExpressionResult2 getUnspecifiedExpressResult(ExpressionResult2 result) {
+        GenericTerm unspecified = ontologyRepository.getTermByNameActive(Term.UNSPECIFIED, Ontology.ANATOMY);
+        Session session = HibernateUtil.currentSession();
+        Query<ExpressionResult2> query = session.createQuery("""
+            from ExpressionResult2 where
+            expressionFigureStage.expressionExperiment = :expressionExperiment
+            AND expressionFigureStage.startStage = :start
+            AND expressionFigureStage.endStage = :end
+            AND superTerm = :superTerm
+            AND expressionFound = :expFound
+            """, ExpressionResult2.class);
+        query.setParameter("expressionExperiment", result.getExpressionFigureStage().getExpressionExperiment());
+        query.setParameter("start", result.getExpressionFigureStage().getStartStage());
+        query.setParameter("end", result.getExpressionFigureStage().getEndStage());
+        query.setParameter("superTerm", unspecified);
+        query.setParameter("expFound", result.isExpressionFound());

```

Comment:
I'm sure you've worked out the logic here, but I'm not sure why the `"expressionFound", true` gets changed to `"expFound", result.isExpressionFound()`

Reply:
Thanks for this catch. I will undo this change from the original one.